### PR TITLE
Add elasticsearch deployment template

### PR DESCRIPTION
**Pull request by 4n4nd**

### CODE REVIEW

File: openshift/lad-elasticsearch-deployment.yaml [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+        metadata:
+          labels:
+            deploymentConfig: ${APPLICATION_NAME}
+            app: ${APPLICATION_NAME}
+        spec:
+          containers:
+            - env:
+                - name: "discovery.type"
+                  value: "single-node"
+              name: ${APPLICATION_NAME}
```
Comment (zmhassan) [inline openshift/lad-elasticsearch-deployment.yaml]: Make `discovery.type` this configurable. Pass in parameters through openshift

Comment (zmhassan) [reply openshift/lad-elasticsearch-deployment.yaml]: default to `single-node`

**Comment by zmhassan** [comment]
@4n4nd Please rebase. LGTM. I'll merge once you do that.

**Comment by 4n4nd** [comment]
I have resolved the conflicts, please squash and merge

**Comment by zmhassan** [comment]
@4n4nd Just was going to suggest its always good practice to keep PR commits to 1. I don't think we can squash your commits. Do you mind squashing and I would be glad to merge.

**Comment by zmhassan** [comment]
Nevermind the bot is at work. Guess it will show as 2 extra commits. Let's avoid this in future.

**Pull request merged**